src/Cake.Core/ICakeEnvironment.cs

```diff
@@ -40,6 +40,13 @@ public interface ICakeEnvironment
         /// <returns>The value of the environment variable.</returns>
         string GetEnvironmentVariable(string variable);
 
+        /// <summary>
+        /// Expands the environment variables in the text. Variables shoud be quoted with %.
+        /// </summary>
+        /// <param name="text">A string containing the names of zero or more environment variables.</param>
+        /// <returns>A string with each environment variable replaced by its value.</returns>
+        string ExpandEnvironmentVariables(string text);
```

Comment:
Been trying this out, and while I really like the feature, we're really hesitant to introduce a breaking change in Cake.Core 0.27.0 - which adding an interface member is.

Been giving some thought on how not to introduce a breaking change, and only possibility would probably be to switch from an interface member to an extension method, something like

`public static string ExpandEnvironmentVariables(this ICakeEnvironment environment, string text)`

Would make it harder to differentiate between Fake and non fake environments, one naive way would be `if (environment is CakeEnvironment)` or just use the same implementation for both (_that way unit test would run same code as scripts_).

`_environmentVariables` could be replaced with `ICakeEnvironment.GetEnvironmentVariables()` in that method.

Reply:
Can you give me some insight why is it a breaking change to add a method to an interface? Meanwhile i will check how to do the same with extension method.

Reply:
An interface is an contract, any addin / module / assembly inheriting that would break as they would no longer uphold the contract.

Reply:
I didn't know that the addins can implement the ICakeEnvironment interface. When is this a valid use case?
I thought ICakeEnvironment is something what is used by the addins and a new feature won't be a breaking change. And you have to update the Cake reference in the addin only if you would like to use that new feature.

Reply:
For the extension method:
what if i don't use the `Environment.ExpandEnvironmentVariables`method but we use the `GetEnvironmentVariables()` of the `ICakeEnvironment` so it will work for normal and fake environment as well. The implementation can be almost the same what is in the `FakeEnvironment.xpandEnvironmentVariables` now.
What do you think? I think this is a better approach than casting the interface to exact types.

Reply:
Issue is that it's a public interface, as such we can't know who inherits from it. Someone might replace it with a module or i.e. proxy it within in a addin or their unit tests etc. 

We try really hard to keep both source code and binary compatibility, and that's why we're very careful about changes in Cake.Core, sometimes the feature justifies a breaking change, but in this case I think it's worth the extra effort to not break existing things.

There's some good guidance on the topic provided by the .NET team at:
https://github.com/dotnet/corefx/tree/master/Documentation/coding-guidelines

Reply:
> For the extension method:
what if i don't use the Environment.ExpandEnvironmentVariablesmethod but we use the GetEnvironmentVariables() of the ICakeEnvironment so it will work for normal and fake environment as well. The implementation can be almost the same what is in the FakeEnvironment.xpandEnvironmentVariables now.
What do you think? I think this is a better approach than casting the interface to exact types.

Sounds reasonable to me.